static/js/reducers/comments.js

```diff
@@ -31,9 +31,23 @@ const appendCommentToTree = (tree: Array<Comment>, comment: Comment, commentId?:
     : R.prepend(comment, tree)
 }
 
+const updateCommentTree = (tree: Array<Comment>, updatedComment: Comment): Array<Comment> => {
+  return R.map((treeComment: Comment) => {
+    return updatedComment.id === treeComment.id
+      ? {
+        ...updatedComment,
+        replies: treeComment.replies
+      }
+      : {
+        ...treeComment,
+        replies: updateCommentTree(treeComment.replies, updatedComment)
+      }
+  }, tree)
+}
```

Comment:
it looks like this function will result in creating a new object for every single comment. is that what we want to do? I'm a little worried about the performance on low-memory devices when we start having to deal with a lot of comments. Maybe my fears are unfounded though?